themes/tiddlywiki/vanilla/base.tid

```diff
@@ -1576,9 +1580,31 @@ html body.tc-body.tc-single-tiddler-window {
 	overflow: auto;
 }
 
-.tc-tiddler-preview-preview {
-	float: right;
-	width: 49%;
+.tc-tiddler-edit-frame .tc-dropzone-editor > div > div {
+	display: grid;
+}
+
+.tc-tiddler-edit-frame .tc-dropzone-editor > div > div.tc-tiddler-preview {
+	grid-template-areas: 
+	"toolbar toolbar"
+	"editor preview";
+	grid-template-columns: 1fr 1fr;
+	grid-template-rows: auto 1fr;
+}
+
+.tc-tiddler-edit-frame .tc-dropzone-editor > div > div:not(.tc-tiddler-preview) {
+	grid-template-areas: 
+	"toolbar"
+	"editor";
+	grid-template-columns: 1fr;
+	grid-template-rows: auto 1fr;
+}
+
+.tc-tiddler-edit-frame .tc-dropzone-editor > div > div.tc-tiddler-preview > .tc-tiddler-preview-preview {
```

Comment:
I do have a problem with all those CSS rulses. IMO they will fail in the future

Reply:
you mean if the DOM structure changes?

Reply:
I'm now using not so brittle CSS selectors. Could you have a look @pmario ?